cfme/tests/cloud/test_providers.py

```diff
@@ -618,12 +646,63 @@ def test_regions_gov_azure(provider):
         setup: Check the region list when adding a Azure Provider.
         startsin: 5.7
     """
-    view = navigate_to(AzureProvider, "Add")
-    # prefill the provider type to enable regions dropdown
-    view.fill({'prov_type': provider.type.capitalize()})
-    available_regions = [opt.text for opt in view.region.all_options]
     # no government regions should be available by default
-    assert not any(reg for reg in available_regions if 'gov' in reg.lower())
+    assert not _azure_has_available_gov_regions()
+
+
+@test_requirements.azure
+@pytest.mark.provider([AzureProvider], scope="function", selector=ONE)
+@pytest.mark.tier(2)
+def test_regions_all_azure(_azure_available_regions_backend, _azure_available_regions):
+    """
+    Need to validate the list of regions we show in the UI compared with
+    regions.rb  Recent additions include UK South
+    These really don"t change much, but you can use this test case id
+    inside bugzilla to set qe_test flag.
+
+    Polarion:
+        assignee: anikifor
+        casecomponent: Cloud
+        caseimportance: medium
+        initialEstimate: 1/12h
+        startsin: 5.6
+    """
+    d = set(_azure_available_regions_backend).symmetric_difference(set(_azure_available_regions()))
+    assert bool(d), f"Regions {list(d)} differ in UI and backend"
+
+
+@test_requirements.azure
+@pytest.mark.provider([AzureProvider], scope="function", selector=ONE)
+@pytest.mark.automates([BZ(1412355)])
+@pytest.mark.tier(2)
+def test_regions_disable_azure(appliance, _azure_has_available_gov_regions, request, provider):
+    """
+    CloudForms should be able to enable/disable unusable regions in Azure,
+    like the Government one for example.
+
+    Polarion:
+        assignee: anikifor
+        casecomponent: Cloud
+        caseimportance: medium
+        initialEstimate: 1/10h
+        testSteps:
+            1. Go into advanced settings and add or remove items from the following section.
+               :ems_azure:
+               :disabled_regions:
+               - usgovarizona
+               - usgoviowa
+               - usgovtexas
+               - usgovvirginia
+    Bugzilla:
+        1412355
+    """
+    assert not _azure_has_available_gov_regions()
+    old = appliance.get_disabled_regions(AzureProvider)
+    appliance.set_disabled_regions(AzureProvider, [])
+    request.addfinalizer(lambda: appliance.set_disabled_regions(AzureProvider, *old))
+    wait_for(_azure_has_available_gov_regions, fail_condition=lambda x: not x, delay=10, timeout=80)
```

Comment:
Is the fail condition necessary here? Can you just do:

```suggestion
    wait_for(lambda: not _azure_has_available_gov_regions(), delay=10, timeout=80)
```

Reply:
oh, I didn't realize that when func result is False => fail
looks cleaner with your suggestion:)